File: csharp/demo/demo02a-join.cs
/*
 * THREAD JOINS
 */
using System;
using System.Threading;



class Demo02A : IRunnable
{
    public void run()
    {
        Thread th = new Thread(doHeavyTask);

        th.Start();
        th.Join();

        Console.WriteLine("Good bye!");
    }

    private void doHeavyTask() {
        // Do a heavy task, which takes a little time
        for (int i = 0; i < 2000000000; ++i);

        Console.WriteLine("Done!");
    }
}

File: python/demo02a_join.py
'''
THREAD JOINS
'''

import threading



def do_heavy_task():
    # Do a heavy task, which takes a little time
    for _ in range(0, 2 * 10**8):
        pass

    print('Done!')



th = threading.Thread(target=do_heavy_task)

th.start()
th.join()

print('Good bye!')
